ui/src/pages/ChallengePage.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect } from "react";
+import "../style/ChallengePage.css";
+import * as core from "@material-ui/core";
+
+import PropTypes from "prop-types";
+import Terminal from "../components/Terminal";
+
+import gfm from "remark-gfm";
+import ReactMarkdown from "react-markdown";
+import useFetchAuth from "../useFetchAuth";
+import Spinner from "../components/Spinner";
+import GenericErrorPage from "./error-pages/genericErrorPage";
+import {useParams} from "react-router-dom";
+
+function a11yProps(index) {
+	return {
+		id: `scrollable-auto-tab-${index}`,
+		"aria-controls": `scrollable-auto-tabpanel-${index}`,
+	};
+}
+
+function TabPanel(props) {
+	const { children, value, index, ...other } = props;
+	return (
+		<div
+		role="tabpanel"
+		hidden={value !== index}
+		id={`scrollable-auto-tabpanel-${index}`}
+		aria-labelledby={`scrollable-auto-tab-${index}`}
+		{...other}
+		>
+		<core.Box style={{ padding: 0 }} p={3}>
+			<core.Typography>{children}</core.Typography>
+		</core.Box>
+		</div>
+	);
+}
+
+TabPanel.propTypes = {
+	children: PropTypes.node,
+	index: PropTypes.any.isRequired,
+	value: PropTypes.any.isRequired,
+};	
+
+export default function ChallengePage() {
+	//value variable control which is the chosen/selected tab in appbar
+	const [value, setValue] = React.useState(0);
+	const handleChange = (event, newValue) => {
```

Comment:
maybe add some comments about how handle change works.

Reply:
The handleChange is just a function to update the value and the API path when an item is clicked on TabMode